pylint/checkers/classes.py

```diff
@@ -1788,6 +1879,74 @@ def _check_unexpected_method_signature(self, node):
                 node=node,
             )
 
+    @staticmethod
+    def _is_wrapped_type(node, type_):
+        return (
+            isinstance(node, astroid.Instance)
+            and node.name == type_
+            and not isinstance(node, astroid.Const)
+        )
+
+    @staticmethod
+    def _is_int(node):
+        if SpecialMethodsChecker._is_wrapped_type(node, "int"):
+            return True
+
+        if isinstance(node, astroid.Const) and isinstance(node.value, int):
```

Comment:
Can simply these three lines to:
```suggestion
      return isinstance(node, astroid.Const) and isinstance(node.value, int)
```